go/processor/floria/calls.go

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2024 Fantom Foundation
+//
+// Use of this software is governed by the Business Source License included
+// in the LICENSE file and at fantom.foundation/bsl11.
+//
+// Change Date: 2028-4-16
+//
+// On the date above, in accordance with the Business Source License, use of
+// this software will be governed by the GNU Lesser General Public License v3.
+
+package floria
+
+import (
+	"fmt"
+
+	"github.com/Fantom-foundation/Tosca/go/tosca"
+)
+
+func Call(interpreter tosca.Interpreter, transaction tosca.Transaction, context tosca.TransactionContext) (tosca.Result, error) {
+
+	blockParameters := tosca.BlockParameters{}
+
+	transactionParameters := tosca.TransactionParameters{
+		Origin:     transaction.Sender,
+		GasPrice:   transaction.GasPrice,
+		BlobHashes: []tosca.Hash{},
+	}
+
+	codeHash := context.GetCodeHash(*transaction.Recipient)
+	code := context.GetCode(*transaction.Recipient)
+
+	params := tosca.Parameters{
+		BlockParameters:       blockParameters,
+		TransactionParameters: transactionParameters,
+		//Context:               runContext, todo implement
+		Kind:      tosca.Call,
+		Static:    false,
+		Depth:     0, // todo add depth check
+		Gas:       transaction.GasLimit,
+		Recipient: *transaction.Recipient,
+		Sender:    transaction.Sender,
+		Input:     transaction.Input,
+		Value:     transaction.Value,
+		CodeHash:  &codeHash,
+		Code:      code,
+	}
+
+	snapshot := context.CreateSnapshot()
+	if err := transferValue(transaction, context); err != nil {
+		context.RestoreSnapshot(snapshot)
+		return tosca.Result{}, err
```

Comment:
nit: Not sure if a failed transaction should result in a cal error, since it might be considered normal operation.

Reply:
In [geth](https://github.com/ethereum/go-ethereum/blob/cf0378499f1bcae65c093c58cd6ca8225e91b125/core/vm/evm.go#L203), the call returns with a insufficient balance error if there is not enough balance.

Reply:
We have been derivating from the `geth` implementation in the past as well if it lead to cleaner interface definitions -- in particular regarding error handling. As we discussed in the past, signaling the result of perfectly acceptable operations through errors is not good practice.

When we say we have to be `geth` compatible and use `geth` as a reference, this is limited to the observable behavior. We are free to change internal aspects as needed.

The decision how to handle insufficient funds for a value transfer will be more clear once this code is reused for nested calls. When covering this, a consistent setup between the [Processor](https://github.com/Fantom-foundation/Tosca/blob/31731009a10dfe16344549cafbd26d25c0763d02/go/tosca/processor.go#L22), [Interpreter](https://github.com/Fantom-foundation/Tosca/blob/31731009a10dfe16344549cafbd26d25c0763d02/go/tosca/interpreter.go#L32), and [RunContext](https://github.com/Fantom-foundation/Tosca/blob/31731009a10dfe16344549cafbd26d25c0763d02/go/tosca/interpreter.go#L77) interfaces should be found and documented.